astropy/io/ascii/cds.py

```diff
@@ -164,14 +210,439 @@ def get_cols(self, lines):
+        for i, col in enumerate(self.cols):
+            # If col is a ``Column`` object but its values are ``SkyCoord`` objects,
+            # convert the whole column to ``SkyCoord`` object, which helps in applying
+            # SkyCoord methods directly.
+            if not isinstance(col, SkyCoord) and isinstance(col[0], SkyCoord):
+                col = SkyCoord(col)
+
+            # Replace single ``SkyCoord`` column by its coordinate components.
+            if isinstance(col, SkyCoord):
+                # If coordinates are given in RA/DEC, divide each them into hour/deg,
+                # minute/arcminute, second/arcsecond columns.
+                if 'ra' in col.representation_component_names.keys():
+                    ra_col, dec_col = col.ra.hms, col.dec.dms
+                    coords = [ra_col.h, ra_col.m, ra_col.s,
+                              dec_col.d, dec_col.m, dec_col.s]
+                    names = ['RAh', 'RAm', 'RAs', 'DEd', 'DEm', 'DEs']
+                    coord_units = [u.h, u.min, u.second,
+                                   u.deg, u.arcmin, u.arcsec]
+                    coord_descrip = ['Right Ascension (hour)', 'Right Ascension (minute)',
+                                     'Right Ascension (second)', 'Declination (degree)',
+                                     'Declination (arcmin)', 'Declination (arcsec)']
+                    for coord, name, coord_unit, descrip in zip(
+                        coords, names, coord_units, coord_descrip):
+                            # Have Sign of Declination only in the DEd column.
+                            if name in ['DEm', 'DEs']:
+                                coord_col = Column(list(np.abs(coord)), name=name,
+                                                   unit=coord_unit, description=descrip)
+                            else:
+                                coord_col = Column(list(coord), name=name, unit=coord_unit,
+                                                   description=descrip)
+                            self.cols.append(coord_col)
+
+                # For all other coordinate types, simply divide into two columns
+                # for latitude and longitude resp. with the unit used been as it is.
+                else:
+                    # Galactic coordinates.
+                    if col.name == 'galactic':
+                        lon_col = Column(col.l, name='GLON',
+                                         description='Galactic Longitude',
+                                         unit=col.representation_component_units['l'])
+                        lat_col = Column(col.b, name='GLAT',
+                                         description='Galactic Latitude',
+                                         unit=col.representation_component_units['b'])
+                    # Ecliptic coordinates, can be any of various available.
+                    else:
+                        lon_col = Column(col.lon, name='ELON',
```

Comment:
Need review by someone who is better with coordinates than I am. This "else" catches all other coordinates. What about e.g. cartesian coordinates?

Reply:
I can add a check for `SkyCoord` objects that have `SphericalRepresentation`.
All other coordinate columns can simply be rendered as string values within a single column.